src/editor/Model/WindowManager.re

```diff
@@ -0,0 +1,48 @@
+open Revery_UI;
+open Oni_Core;
+
+type layout =
```

Comment:
One concern I have is this layout - it seems like it will be limiting. In particular, it won't allow for nested splits, or multiple split levels.

I think a more natural representation of the splits is a `tree` representation:
```
type splitDirection =
| Horizontal
| Vertical;

type splitTree =
| Parent(direction, list(splitTree))
| Leaf(split)
```

This allows for nesting of splits. You could have a side-by-side layout of two editors, like:
```
Parent(Horizontal, [Leaf(editor1), Leaf(editor2)]
```
And then maybe you decide you want to split editor2 into a vertical split:
```
Parent(Horizontal, [Leaf(editor1), Parent(Vertical, [Leaf(editor2), Leaf(editor3)])
```
etc...

If we use a tree to model the splits - we can open up having nesting.

However, in that case, we might want a separate model for sidebars, like a dock:
```
type dock = list(split)
```

Then our Window Manager might contain:
```
type t = {
   leftDock: dock,
   rightDock: dock,
   windows: splitTree,
};
```